pkg/sources/gitlab/gitlab.go

```diff
@@ -262,119 +263,89 @@ func (s *Source) getRepos() ([]*url.URL, []error) {
 
 func (s *Source) scanRepos(ctx context.Context, chunksChan chan *sources.Chunk, repos []*url.URL) []error {
 	wg := sync.WaitGroup{}
-	errs := []error{}
+	var errs []error
 	var errsMut sync.Mutex
-	if s.authMethod == "UNAUTHENTICATED" {
-		for i, u := range repos {
-			if common.IsDone(ctx) {
-				// We are returning nil instead of the scanErrors slice here because
-				// we don't want to mark this scan as errored if we cancelled it.
-				return nil
+	errs = s.scan(ctx, chunksChan, repos, &wg, &errsMut, errs)
+	wg.Wait()
+
+	return errs
+}
+
+func (s *Source) scan(ctx context.Context, chunksChan chan *sources.Chunk, repos []*url.URL, wg *sync.WaitGroup, errsMut *sync.Mutex, errs []error) []error {
```

Comment:
**Nit/Question:** Its very possible I'm overlooking something, but it seems like it would be easier to keep `scan` and `scanRepos` as one since `errs` and `errsMut` are being passed into `scan`

Reply:
Glad you mentioned this, yea it actually now makes more sense within the original function. When i first broke it apart there was still a good amount of code within `scanRepos` that warranted a separate function for the actual scanning portion. Now i agree with you it can be condensed.
Combined back into a single fxn.